src/main/kotlin/day17/utils/CommonUtils.kt

```
@@ -0,0 +1,15 @@
+package day17.utils
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+fun convertToDate(dateString: String?): LocalDate {
+    return try {
+        dateString?.let { date ->
+            val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+            LocalDate.parse(date, formatter)
+        } ?: LocalDate.now()
+    } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)